src/components/modebar/buttons.js

```diff
@@ -183,7 +183,8 @@ function handleCartesian(gd, ev) {
         var mag = (val === 'in') ? 0.5 : 2,
             r0 = (1 + mag) / 2,
             r1 = (1 - mag) / 2,
-            axList = Axes.list(gd, null, true);
+            axList = Axes.list(gd, null, true),
+            allEnabled = 'on';
```

Comment:
I agree, this would be better as a boolean, and is how I had it originally. I wound up changing it to take advantage of the modebar logic which sets the `active` class. Because the `val` on the button is passed through a `data-val` attr on the dom, and the is a strict equality check at https://github.com/plotly/plotly.js/blob/master/src/components/modebar/modebar.js#L211 I found I couldn't use a boolean. Open to suggestions that let me keep this as a boolean, but get to use the state display on the modebar.